### DISCUSSION TITLE
Issues with PostJsonAsync using System.Text.Json

### DISCUSSION THREAD

**Discussion by CoolDaddyCool** [dotnet/aspnetcore#50332] [Q&A] [answered]
Hello there,
I have a simple class
    public class Player
    {
        [JsonPropertyName("id")]
        public int Id{ get; set; }

        [JsonPropertyName("name")]
        public int Name{ get; set; }
}
When I try to post as
`
string stringJson = JsonSerializer.Serialize(Player);'
var content = new StringContent(stringJson, System.Text.Encoding.UTF8, "application/json");
var response = await httpClient.PostAsync(“myurl”, content);
`
it works fine.
But when I try to post as:
`
var response = await httpClient.PostAsJsonAsync(“myurl”, Player);
`
I get BadRequest.
I'm not the expert, but I think those 2 approaches should have the same result. Can anyone please try to explain why I get BadRequest with PostAsJsonAsync?

**Comment by davidfowl** [comment]
Where is the server code?

**Comment by campersau** [comment, marked as answer]
`HttpClient.PostAsJsonAsync` defaults to [`JsonSerializerDefaults.Web` serializer options](https://github.com/dotnet/runtime/blob/71c68bf465ffbb5abb3706405717325f34ca504f/src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/JsonHelpers.cs#L15) which are different from using `JsonSerializer.Serialize` directly.

See:
https://github.com/dotnet/runtime/blob/71c68bf465ffbb5abb3706405717325f34ca504f/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs#L142-L147

**Reply by CoolDaddyCool** [reply]
Thank you very much
